src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -432,16 +577,26 @@ def rmatmul(self, y):
         """
         raise NotImplementedError
 
-    def xor(self, y):
+    def xor(self, y: Union[int, torch.Tensor, "ReplicatedSharedTensor"]):
         """Apply the "xor" operation between "self" and "y".
 
         Args:
-            y: self^y
+            y: public bit
+
+        Returns:
+            ReplicatedSharedTensor: Result of the operation.
 
         Raises:
-            NotImplementedError: Raised when implementation not present
-        """
-        raise NotImplementedError
+            ValueError : If ring size is invalid.
+        """
+        if self.ring_size == 2:
+            return self + y
```

Comment:
Q: Shouldn't we check the values of "y"? (what happens if "y" is 132, shouldn't we raise an error?)

Reply:
y is a public bit, also now user how to check ring size if its just a tensor

Reply:
@gmuraru also since we are using + and other ops it will got to apply ops and sanity checks will be done, so i dont think we need it here.